### ISSUE TITLE
HACS Teufel integration not connecting to Teufel host

### ISSUE THREAD

**Issue by tvrhomie** [B5r1oJ0A9G/teufel_raumfeld#16]
I run Homeassistant core-2021.6.6 on a QNAP Docker station.
I've installed HACS and managed to install the teufel integration.
When adding the integration to Home Asisstant the Teufel Host does not connect (not when using the IP of my raumfeld expand nor any of the raumfeld speakers of which I have a pair of series L, 3 OneS speakers and the soundbar.)
--> when entering any of the speaker - IP addresses I get a fail to connect error
--> when enetering the IP address of the raumfeld connect there is no error but the integration seems to endlessly process

Any guidance to complete the TEufel integration to my HA would be appreciated. I'm quite a newbie. 
I've read the other issues and posts and think I miss the Hassfeld install. Can you confirm that to be the issue and if so, 
I understand Hassfeld is another integration on Home Assistant. Should I install and how to do that on HA Core?

**Comment by B5r1oJ0A9G** [comment]
Hello,

There is always only one Host device in a Raumfeld network. The integration has to be configured with this host. From what you describe I guess "raumfeld connect" is your Raumfeld host. Configuring other devices than the actual Host device is supposed to fail.

It is usually not needed to install ```hassfeld``` manually. HACS should install it automatically alongside to ```teufel_raumfeld``` as dependency. By design a Home Assistent integration must not include any device related logic. If device related logic is needed, this has to be separated in another module. ```teufel_raumfeld``` is the integration and ```hassfeld``` includes all the Teufel Raumfeld related logic. ```teufel_raumfeld``` is dedicated to Home Assistant while ```hassfeld``` can also be used in any Python program to interact with, resp. to manage Teufel Raumfeld devices.

My first suggestion would be to enable debug logging for ```teufel_raumfeld``` and to watch out for suspicious messages, resp. to report it here for review. See https://github.com/B5r1oJ0A9G/teufel_raumfeld/issues/2#issuecomment-780948127.

**Comment by tvrhomie** [comment]
Thanks @B5r1oJ0A9G / Marc I will set the log to debug and share the outcomes.

**Comment by tvrhomie** [comment]
I added the debug code to the configuration.yaml
This did not generate a raumfeld.wsd logging, the logging from Home assistant is here:
2021-07-07 20:41:09 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration teufel_raumfeld which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-07-07 20:41:09 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-07-07 20:41:09 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration config_check which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-07-07 20:41:09 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration smartthinq_sensors which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-07-07 20:41:23 WARNING (MainThread) [custom_components.teufel_raumfeld] You are using teufel_raumfeld, which is still in alpha phase and therefore subject to change. This includes, among other things, the addition, redesign or removal of functionality.
2021-07-07 20:41:23 DEBUG (MainThread) [custom_components.teufel_raumfeld] __init__.py->async_setup: OrderedDict([('default_config', {}), ('tts', [OrderedDict([('platform', 'google_translate')])]), ('group', {}), ('automation', {}), ('script', OrderedDict([('check_configuration', OrderedDict([('alias', 'Check Configuration'), ('sequence', [OrderedDict([('service', 'homeassistant.check_config')])]), ('mode', 'single')]))])), ('scene', {}), ('logger', OrderedDict([('default', 'warn'), ('logs', OrderedDict([('custom_components.teufel_raumfeld', 'debug')]))]))])
2021-07-07 20:41:25 DEBUG (MainThread) [custom_components.teufel_raumfeld] Setting logging level of hassfeld to: DEBUG
2021-07-07 20:41:25 DEBUG (MainThread) [hassfeld] __init__.py->__init__: Session for aiohttp requests was passed.
2021-07-07 20:41:25 DEBUG (MainThread) [custom_components.teufel_raumfeld] Setting logging level of hassfeld to: DEBUG
2021-07-07 20:41:25 DEBUG (MainThread) [hassfeld] __init__.py->__init__: Session for aiohttp requests was passed.
2021-07-07 20:41:25 DEBUG (MainThread) [custom_components.teufel_raumfeld] Setting logging level of hassfeld to: DEBUG
2021-07-07 20:41:25 DEBUG (MainThread) [hassfeld] __init__.py->__init__: Session for aiohttp requests was passed.
2021-07-07 20:41:25 DEBUG (MainThread) [custom_components.teufel_raumfeld] Setting logging level of hassfeld to: DEBUG
2021-07-07 20:41:26 WARNING (MainThread) [homeassistant.setup] Setup of timer is taking over 10 seconds.
2021-07-07 20:41:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Starting web service update coroutine
2021-07-07 20:41:27 DEBUG (MainThread) [hassfeld] __init__.py->async_update_all: Session for webservice requests was passed.
2021-07-07 20:41:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Starting web service update coroutine
2021-07-07 20:41:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Starting web service update coroutine
2021-07-07 20:41:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Starting web service update coroutine
2021-07-07 20:41:27 DEBUG (MainThread) [hassfeld] __init__.py->async_update_all: Session for webservice requests was passed.
2021-07-07 20:41:27 DEBUG (MainThread) [hassfeld] __init__.py->async_update_all: Session for webservice requests was passed.
2021-07-07 20:41:27 DEBUG (MainThread) [hassfeld] __init__.py->async_update_all: Session for webservice requests was passed.
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: host_info
2021-07-07 20:41:30 CRITICAL (MainThread) [hassfeld] Long-polling failed with error: <class 'TypeError'>list indices must be integers or slices, not str
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: host_info
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:41:31 CRITICAL (MainThread) [hassfeld] Long-polling failed with error: <class 'TypeError'>list indices must be integers or slices, not str
2021-07-07 20:41:31 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: host_info
2021-07-07 20:41:31 CRITICAL (MainThread) [hassfeld] Long-polling failed with error: <class 'TypeError'>list indices must be integers or slices, not str
2021-07-07 20:41:31 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: host_info
2021-07-07 20:41:31 CRITICAL (MainThread) [hassfeld] Long-polling failed with error: <class 'TypeError'>list indices must be integers or slices, not str
2021-07-07 20:41:37 WARNING (MainThread) [homeassistant.config_entries] Config entry 'ExperiaBox-V10' for upnp integration not ready yet: None; Retrying in background
2021-07-07 20:42:24 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: teufel_raumfeld
2021-07-07 20:43:24 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: teufel_raumfeld
2021-07-07 20:44:24 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: teufel_raumfeld
2021-07-07 20:45:24 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: teufel_raumfeld
2021-07-07 20:46:16 WARNING (MainThread) [homeassistant.bootstrap] Setup timed out for stage 2 - moving forward
2021-07-07 20:46:26 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:46:26 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:46:26 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:46:26 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:46:26 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:46:26 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:46:26 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:46:26 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:46:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:46:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:46:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:46:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:46:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:46:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices
2021-07-07 20:46:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:46:27 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state
2021-07-07 20:47:01 INFO (MainThread) [hassfeld] Long-polling timed out
2021-07-07 20:47:01 INFO (MainThread) [hassfeld] Long-polling timed out
2021-07-07 20:47:01 INFO (MainThread) [hassfeld] Long-polling timed out
2021-07-07 20:47:01 INFO (MainThread) [hassfeld] Long-polling timed out
2021-07-07 20:47:01 INFO (MainThread) [hassfeld] Long-polling timed out
2021-07-07 20:47:01 INFO (MainThread) [hassfeld] Long-polling timed out
2021-07-07 20:47:01 INFO (MainThread) [hassfeld] Long-polling timed out
2021-07-07 20:47:01 INFO (MainThread) [hassfeld] Long-polling timed out

In the process I tried to remove and re-install the integration, 

What also might be interesting:

The HA-container in QNAP provides this error log: 
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices                                      
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state                                 
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: devices                                      
2021-07-07 20:41:30 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: system_state                                 
2021-07-07 20:41:31 CRITICAL (MainThread) [hassfeld] Long-polling failed with error: <class 'TypeError'>list indices must be integers or slices, not str 
2021-07-07 20:41:31 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: host_info                                    
2021-07-07 20:41:31 CRITICAL (MainThread) [hassfeld] Long-polling failed with error: <class 'TypeError'>list indices must be integers or slices, not str 
2021-07-07 20:41:31 DEBUG (MainThread) [custom_components.teufel_raumfeld] Update event triggered for type: host_info                                    
2021-07-07 20:41:31 CRITICAL (MainThread) [hassfeld] Long-polling failed with error: <class 'TypeError'>list indices must be integers or slices, not str 
2021-07-07 20:41:37 WARNING (MainThread) [homeassistant.config_entries] Config entry 'ExperiaBox-V10' for upnp integration not ready yet: None; Retrying

**Comment by B5r1oJ0A9G** [comment]
Thanks for the logs @tvrhomie!

Do you happen to have a room with more than one device configured in it, e.g. two Teufel One S?

**Comment by tvrhomie** [comment]
Hi Marc,

Yes I do, my setup is:
1) TV area: Teufel soundbar (cable connected to the network)
2) living room: two Teufel L (which are connected as one speaker and a
child - wifi connected directly to Teufel Host))
3) Kitchen: one Teufel OneS (wifi connected to the network)
4) Office: *two* teufel One S (which are individually connected to Wifi to
the network)

Op do 15 jul. 2021 om 21:13 schreef Marc ***@***.***>:

**Comment by tvrhomie** [comment]
I've split the two speakers in the office to two different rooms and resinstalled the entire integration. Problem solved

**Comment by B5r1oJ0A9G** [comment]
Thanks for the feedback @tvrhomie. This is unfortunately a known limitation. A quick fix was made but it is not tested yet.

Duplicate of  #13
